Octokit.Tests/Clients/AuthorizationsClientTests.cs

```diff
@@ -290,9 +290,9 @@ public async Task ChecksApplicationAuthenticateAtCorrectUrl()
 
                 authEndpoint.CheckApplicationAuthentication("clientId", "accessToken");
 
-                client.Received().Get<ApplicationAuthorization>(
-                    Arg.Is<Uri>(u => u.ToString() == "applications/clientId/tokens/accessToken"),
-                    null);
+                client.Received().Post<ApplicationAuthorization>(
+                    Arg.Is<Uri>(u => u.ToString() == "applications/clientId/token"),
+                    Args.Object);
```

Comment:
Can we assert that the object has the expected `access_token` parameter here? I think we have some other examples of this floating around (I'm not sure whether it can be `AccessToken` and is serialized later to `access_token`, or must be `access_token`) but it'd be great to verify the behaviour...

Reply:
Refreshed my memory a bit and it turns out that we don't do any conversion, so you're right to use `access_token` here:

https://github.com/octokit/octokit.net/blob/9b3cf309cf268916d54d73583869e65f95b6e627/Octokit/Clients/AuthorizationsClient.cs#L99-L105

The use of anonymous types however makes testing a bit of a pain. I was able to put together this one-liner to verify it, but I think we can do better on a follow-up PR. Thoughts?

```suggestion
                    Arg.Is<Object>(u => u.GetType().GetProperty("access_token").GetValue(u).ToString() == "accessToken"));
```

Reply:
Yeah, was just going down that rabbit hole. I think to do better we would need a concrete type for the structure that has the access_token field. Happy to add that in a follow up PR. If we do add a concreate type, we should probably make a similar change for other cases where we just use an unadorned Args.Object in the test case. (e.g. TheGetOrCreateApplicationAuthenticationMethod.GetsOrCreatesAuthenticationAtCorrectUrl)

Reply:
Because these are internal types (rather than ones the user creates in code) I'm fine with leaving them as-is. We don't have great test coverage over the parameters sent in `Post`/`Patch`/`Put` but I'm open to improvements here.